### PR TITLE
Use providers array instead providerUrl

### PR DESCRIPTION
**Pull request by andreogle**
## Changes

Replaces `providerUrl` with a `providers` array as outlined in the comment here: https://github.com/api3dao/chains/issues/103#issuecomment-1759807602

Closes: https://github.com/api3dao/chains/issues/103 and https://github.com/api3dao/chains/issues/4

### CODE REVIEW

File: chains/avalanche-testnet.json [hunk 1]
```diff
@@ -4,7 +4,12 @@
   "id": "43113",
   "symbol": "testAVAX",
   "testnet": true,
-  "providerUrl": "https://api.avax-test.network/ext/bc/C/rpc",
+  "providers": [
+    {
+      "alias": "avax-test",
```
Comment (andreogle) [inline chains/avalanche-testnet.json:9]: I just used the domain (without TLD) as the alias. Let me know if you want any of these changed

File: src/types.ts [hunk 2]
```diff
@@ -15,14 +15,26 @@ export const chainExplorerSchema = z.object({
   browserUrl: z.string().url(),
 });
 
+export const chainProviderSchema = z
+  .object({
+    alias: z.string(),
+    homepageUrl: z.string().url().optional(),
+    rpcUrl: z.string().url().optional(),
+  })
+  .refine(
```
Comment (andreogle) [inline src/types.ts:24]: Note that either `homepageUrl` or `rpcUrl` is required. Having providers that only have an `alias` field feels like a mistake

File: chains/avalanche-testnet.json (hunk 1 above)
Comment (bbenligiray) [reply chains/avalanche-testnet.json:9]: I think we should consider using `default` and `public` wherever applicable as per https://github.com/wagmi-dev/viem/blob/5c95fafceffe7f399b5b5ee32119e2d78a0c8acd/src/types/chain.ts#L57

Comment (andreogle) [reply chains/avalanche-testnet.json:9]: Ah yeah - I like that. Do you have a preference for making the existing ones `default` or `public`?

File: src/types.ts [hunk 3]
```diff
@@ -15,14 +16,51 @@ export const chainExplorerSchema = z.object({
...
+    // Either rpcUrl or homepageUrl must be present
+    (provider) => provider.rpcUrl || provider.homepageUrl,
+    { message: 'rpcUrl or homepageUrl is required' }
+  );
+
+export const chainProvidersSchema = z.array(chainProviderSchema).superRefine((providers, ctx) => {
```
Comment (andreogle) [inline src/types.ts:31]: Some more validations here:

1. Each chain must have at least 1 provider with `alias === 'default'`
2. `alias` values must be unique
3. providers with `alias === 'default'` or `alias === 'public` must also have an `rpcUrl` value

File: chains/avalanche-testnet.json (hunk 1 above)
Comment (andreogle) [reply chains/avalanche-testnet.json]: (I went with `default` - let me know if you disagree)

Comment (bbenligiray) [reply chains/avalanche-testnet.json]: Yup, I had the same preference [here](https://github.com/api3dao/chains/pull/85#discussion_r1376084871)

File: src/types.ts (hunk 2 above)
Comment (bbenligiray) [reply src/types.ts:25]: Agreed

File: src/types.ts (hunk 3 above)
Comment (bbenligiray) [reply src/types.ts:31]: [Agreed](https://github.com/api3dao/chains/pull/85#discussion_r1376078930)

**Review by bbenligiray — APPROVED** [review]
Looking good

**Pull request merged**